internal/controller/controllers/bmh_agent_controller.go

```diff
@@ -694,7 +694,18 @@ func (r *BMACReconciler) reconcileSpokeBMH(ctx context.Context, log logrus.Field
 		return reconcileError{err}
 	}
 
-	_, err = r.ensureSpokeBMHSecret(ctx, log, spokeClient, bmh)
+	key := types.NamespacedName{
+		Namespace: bmh.Namespace,
+		Name:      bmh.Spec.BMC.CredentialsName,
+	}
+	// Make sure that the BMC Credentials Secret has the required label
+	err = ensureSecretLabel(ctx, key.Name, key.Namespace, CustomSecretLabelName, CustomSecretLabelValue, r.Client, r.APIReader)
+	if err != nil {
+		log.WithError(err).Error("error setting label on BMC Credentials Secret")
+		return reconcileError{err}
+	}
```

Comment:
This will query the secret (using the non-cached client) on every reconcile. An alternative implementation would be to modify the `ensureSpokeBMHSecret` function by catching the `NotFound` error and calling `ensureSecretLabel` *only* in that case.

The `ensureSecretLabel` function could return the secret so that we don't have to fetch it twice.

Let me know if this makes sense.

Reply:
Yes, let's please do this. We should only use the non-caching client if we fail to find the Secret in the current cache.

Reply:
Sure